html/controls/sk/sk-sheet.js

```diff
@@ -306,13 +288,29 @@ function prepareTripEditor() {
 		tripEditor.append($('<table>')
 				.append($('<tr>')
 						.append($('<td colspan="3">')
-								.append($('<input type="number" min="0">').attr('id', 'score'))
-								.append($('<input type="checkbox">').attr('id', 'afterSP'))
-								.append($('<span>').addClass('Infotext').text('SP in this or prior trip'))))
+								.append($('<input type="number" min="0">').attr('id', 'score').keydown(function(event) {
+									if (event.which == 13) tripEditor.find('#submit').click(); }).change(function() {
+										WS.Set(tripEditor.data('prefix')+'Score', $(this).val());
+									}))
+								.append($('<button>').attr('id', 'afterSP').text('SP in this or prior trip').button().click(function() {
+									var check = !$(this).hasClass('checked');
+									$(this).toggleClass('checked', check);
+									WS.Set(tripEditor.data('prefix')+'AfterSP', check);
+								}))))
 				.append($('<tr class="buttons">')
-						.append($('<td width="30%">').append($('<button>').attr('id','submit').text('Submit')))
-						.append($('<td width="40%">').append($('<button>').attr('id','insert_before').text('Insert Before')))
-						.append($('<td width="30%">').append($('<button>').attr('id','remove').text('Remove')))));
+						.append($('<td width="30%">').append($('<button>').attr('id','submit').text('Submit').button().click(function() {
+							if (tripEditor.find('#score').val() == '') {
+								WS.Set(tripEditor.data('prefix')+'Remove', true);
+							}
+							tripEditor.dialog('close');
+						})))
+						.append($('<td width="40%">').append($('<button>').attr('id','insert_before').text('Insert Before').button().click(function() {
+							WS.Set(tripEditor.data('prefix')+'InsertBefore', true);
```

Comment:
Insert Before is across two lines for me. For consistency with the jam/period editor, Delete should be first.